rules/terraform_locals_order.go

```diff
@@ -0,0 +1,124 @@
+package rules
+
+import (
+	"fmt"
+	"github.com/hashicorp/hcl/v2"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+	"github.com/terraform-linters/tflint-ruleset-terraform/project"
+)
+
+// TerraformLocalsOrderRule checks whether comments use the preferred syntax
+type TerraformLocalsOrderRule struct {
+	tflint.DefaultRule
+}
+
+// NewTerraformLocalsOrderRule returns a new rule
+func NewTerraformLocalsOrderRule() *TerraformLocalsOrderRule {
+	return &TerraformLocalsOrderRule{}
+}
+
+// Name returns the rule name
+func (r *TerraformLocalsOrderRule) Name() string {
+	return "terraform_locals_order"
+}
+
+// Enabled returns whether the rule is enabled by default
+func (r *TerraformLocalsOrderRule) Enabled() bool {
+	return false
+}
+
+// Severity returns the rule severity
+func (r *TerraformLocalsOrderRule) Severity() tflint.Severity {
+	return tflint.NOTICE
+}
+
+// Link returns the rule reference link
+func (r *TerraformLocalsOrderRule) Link() string {
+	return project.ReferenceLink(r.Name())
+}
+
+// Check checks whether single line comments is used
+func (r *TerraformLocalsOrderRule) Check(runner tflint.Runner) error {
+	files, err := runner.GetFiles()
```

Comment:
You can use `GetLocals()` instead of `GetFiles()`.
https://github.com/terraform-linters/tflint-ruleset-terraform/blob/6bee33cc53dc74b9917f4af5591bbbc08e1e577f/terraform/runner.go#L63

Reply:
Thanks for the tip, I've checked `Local` struct, it contains name and `Range`, which contains file name and range in that file.

What we want to sort is the local argument inside a `locals` block, but not in a file range, since this `Local` struct lacks block information, I think it might be easier to analyze them in File->Block->Attribute way, could I keep this way?

Reply:
That makes sense. In this case, it certainly seems better to analyze by file.